### ISSUE TITLE
Question: How to upgrade an existing project ?

### ISSUE THREAD

**Issue by messenjer** [vuejs/create-vue#70]
Hello,

Is there a recommendation for updating an existing project created with vue-create?  ( `npm init vue@3` ) 

Since the creation of my app, I see that there have been some fixes, like: 

- fix(cypress-ct): avoid cypress types polluting normal source files (#62)
- fix: vitest tsconfig should include all files under src (#56)
- ci: ignore changes in markdown files fix(vitest): allow to typecheck tests (#47)

Thanks

Mathieu

**Comment by sodatea** [comment]
For example, if you created your project two weeks ago, answered yes to "Add TypeScript", "Add JSX Support", and "Add Vitest for Unit Testing", you can run the following commands to see the diffs between the templates generated then and now:

```sh
git clone https://github.com/vuejs/create-vue-templates.git
cd create-vue-templates
git log --since="2 weeks ago" -p typescript-jsx-vitest
# or
# git log --since="2022.02.10" -p typescript-jsx-vitest
```

The https://github.com/vuejs/create-vue-templates repository stores snapshots of the generated templates after each release of `create-vue`.

**Comment by segevfiner** [comment]
Another interesting question would be how to add features to an existing project that you chose not to add them in the first place.

**Comment by messenjer** [comment]
You can try to use `diff -r`

```
git clone https://github.com/vuejs/create-vue-templates.git
cd create-vue-templates
diff -r typescript-pinia-vitest typescript-pinia-vitest-cypress
```

**Comment by mjwfilippo** [comment]
> Another interesting question would be how to add features to an existing project that you chose not to add them in the first place.

has there been an answer to this yet? I cannot seem to find anything, and the course i am taking doesn't help. I am currently not able to continue and don't know what to do

**Comment by bmulholland** [comment]
What about a mode that does a soft run of create-vue on an existing app, and then shows the diffs for each file? I think that would be easy to build, trivial to maintain, and go a long way to helping people stay up to date.

**Comment by bmulholland** [comment]
Yep, cool. My strategy for such things is I invest time in them roughly in line with the inconvenience they cause me, when they cause me that pain. That means I'll watch this project for releases and put work into such a feature as I need to do manual work to stay up to date.

**Comment by cexbrayat** [comment]
👍 

My strategy to be up to date is to do something similar to https://github.com/vuejs/create-vue/issues/70#issuecomment-1193334622

**Comment by bmulholland** [comment]
Oh yeah that's almost exactly what I had in mind. Nice!

I'll probably start with that. One easy approach for the feature I suggested is to basically wrap your solution in the TUI, to auto-fill the dir in the templates repo.
